bay/docker/hosts.py

```diff
@@ -171,7 +173,40 @@ def build_host_ip(self):
         Returns the internal IP of the host as seen from the containers during
         build, which is the gateway of the default bridge network.
         """
-        return self.client.inspect_network("bridge")['IPAM']['Config'][0]['Gateway']
+        """
+        Determines the "host" IP containers should use.
+
+        Get the Gateway IP from the docker daemon.
+        """
+        version = LooseVersion(self.client.version()['Version'].split("-")[0])
+        version_17_06 = LooseVersion("17.06.0")
+        version_17_12 = LooseVersion("17.12.0")
+
+        if sys.platform == "darwin" and version >= version_17_06:
+            # MacOS + recent version of Docker for Mac
+            gateway_ip = "docker.for.mac.host.internal" if version >= version_17_12 else "docker.for.mac.localhost"
```

Comment:
should we rename `gateway_ip` to `host_machine__dns` then?, it's not an IP anymore